source/material_model/latent_heat.cc

```diff
@@ -177,11 +177,11 @@ namespace aspect
               for (unsigned int phase=0; phase<phase_function.n_phase_transitions(); ++phase)
                 {
                   const double depth = this->get_geometry_model().depth(in.position[i]);
-                  const double pressure_depth_derivative = (this->get_adiabatic_conditions().pressure(position) > 0)
+                  const double pressure_depth_derivative = (depth > 0.0)
```

Comment:
I think technically, both the depth and the adiabatic pressure need to be larger than 0 (if one of them is negative, the derivative is negative and doesn't make sense anymore).